packages/core/src/saml-applications/routes/anonymous.ts

```diff
@@ -1,14 +1,34 @@
 import { z } from 'zod';
 
+import RequestError from '#src/errors/RequestError/index.js';
 import koaGuard from '#src/middleware/koa-guard.js';
 import type { AnonymousRouter, RouterInitArgs } from '#src/routes/types.js';
+import assertThat from '#src/utils/assert-that.js';
+
+import {
+  generateAutoSubmitForm,
+  createSamlResponse,
+  handleOidcCallbackAndGetUserInfo,
+  setupSamlProviders,
+} from './utils.js';
+
+const samlApplicationSignInCallbackQueryParametersGuard = z.union([
```

Comment:
Can you double-check the implementation of our node SDK?  Better align the design with other traditional web SDKs.

Reply:
I'm aware of the fact that in a normal flow of OIDC callback page, we also need to validate the consistency of `state` and `redirectUri`. This part will be added after the [PR#6881](https://github.com/logto-io/logto/pull/6881) is ready. I can add TODO for reference.

Reply:
LOG-10602